targets/mbed/source/jerry_extapi.cpp

```diff
@@ -45,39 +45,40 @@ NAME ## _handler (const jerry_api_object_t * function_obj_p __UNSED__, \
 DECLARE_HANDLER(assert)
 {
   if (args_cnt == 1
-      && args_p[0].type == JERRY_API_DATA_TYPE_BOOLEAN
-      && args_p[0].u.v_bool == true)
+      && jerry_value_is_boolean (args[0])
+      && jerry_get_boolean_value (args[0]))
```

Comment:
I think this is a double check, jerry_get_boolean_value will call this jerry_value_is_boolean function, and return with false if it isn't a boolean value.